### ISSUE TITLE
Support Uniqueness on ObjectId fields

### ISSUE THREAD

**Issue by visageae** [matteodelabre/mongoose-beautiful-unique-validation#23]
Hi,
I have several compound indexes and some are like this one : 

```
var clientSchema = new Schema({
  group: {type: Schema.Types.ObjectId, required: true},
  filemd5 : {type: String},
  status: {type: String}
});

clientSchema.index({filemd5: 1, group: 1}, {unique: true});
```

Unfortunately, using your library when it tries to beautify the duplicate message error I get this : 

`[Error: mongoose-beautiful-unique-validation error: cannot parse duplicated values to a meaningful value. Parsing error: Unexpected token O]`

The original error message is the following : 
` { [MongoError: E11000 duplicate key error collection: visage.candidate-submission index: candidateEmail_1_job_1 dup key: { : "md5file", : ObjectId('284028532053528') }]
`

My guess is that it happens because you try to parse the { : "md5file", : ObjectId('284028532053528') } part with as a JSON string. Unfortunately it goes wrong for the ObjectId because this is BSON notation not JSON.
Can we expect a fix for this ?

Thank you.

**Comment by visageae** [comment]
By the way https://www.npmjs.com/package/bson
👍

**Comment by matteodelabre** [comment]
Hey @visageae, thanks for the report! I have to admit that the parsing for the Object section (`{ : "md5file", : ObjectId('284028532053528') }`) is a bit hackish.

I don't know what format it is. Are you sure the `bson` module can parse it? Could you show an example logic to parse it?

**Comment by matteodelabre** [comment]
The current parsing logic is there: https://github.com/matteodelabre/mongoose-beautiful-unique-validation/blob/master/index.js#L72

**Comment by visageae** [comment]
It's a bit more complicated than I thought... It's a subset of BSON and BSON cannot be parsed from this kind of string.... 
Maybe you should consider a quick and dirty fix by writing a regexp that consider all these types https://docs.mongodb.com/manual/reference/bson-types/ as String...

**Comment by matteodelabre** [comment]
We should maybe dig into MongoDB's source to find how this string is being generated.

**Comment by matteodelabre** [comment]
Fixed in version 4.0.0! :tada:
